### DISCUSSION TITLE
queryFnParamsFilter function is never getting called.

### DISCUSSION THREAD

**Discussion by nanandn** [TanStack/query#1123] [General] [answered]
I have a query like this... But queryFnParamsFilter never gets called. Is this is the right way to use it. 
What I am trying to achieve is that I don't want the query, start and size to be cached. I want the cache key to be just "contentItems", but my fetch function needs the other params.

return useQuery({
        queryKey: ["contentItems", { query: q, start, size }],
        queryFn: fetchContentItems,
        queryFnParamsFilter: (params) => {
            console.log(`Params: ${JSON.stringify(params)}`)
            return params;
        }
    });

**Comment by TkDodo** [comment, marked as answer]
The object syntax of `useQuery` looks like this:

```
 // or using the object syntax
 
const queryInfo = useQuery({
  queryKey,
  queryFn,
  config,
})
```

so you would probably need:

```
return useQuery({
    queryKey: ["contentItems", { query: q, start, size }],
    queryFn: fetchContentItems,
    config: {
        queryFnParamsFilter: (params) => {
            console.log(Params: ${JSON.stringify(params)})
            return params;
        }
    }
});
```
